### PR TITLE
DPMMA-2498 Email preheader preview

### PR DESCRIPTION
**Pull request by patrykgruszka**
| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ N ]
| New feature/enhancement? (use the a.x branch)      | [ Y ]
| Deprecations?                          | [ N ]
| BC breaks? (use the c.x branch)        | [ N ]
| Automated tests included?              | [ Y ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 

#### Description:
This PR adds Email text preview (preheader) in the email view page.
image
image

#### Steps to test this PR:

1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Create an email
3. Go to email view
4. Check if the preview text is displayed correctly.

### CODE REVIEW

File: app/bundles/EmailBundle/Tests/Helper/PlainTextHelperTest.php [hunk 1]
```diff
@@ -0,0 +1,110 @@
+<?php
+
```
Comment (tomekkowalczyk) [inline app/bundles/EmailBundle/Tests/Helper/PlainTextHelperTest.php:2]: You can add strict types

File: app/bundles/EmailBundle/Tests/Helper/PlainTextHelperTest.php [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+        $plainTextHelper->setHtml($htmlContent);
+        $actualPlainText = $plainTextHelper->getPreview();
+
+        $this->assertEquals($expectedPlainText, $actualPlainText);
+    }
+
+    /**
+     * @return array<int, array<int, string|int|null>>
+     */
+    public function getPreviewProvider()
```
Comment (tomekkowalczyk) [inline app/bundles/EmailBundle/Tests/Helper/PlainTextHelperTest.php]: return type array

File: app/bundles/EmailBundle/Helper/PlainTextHelper.php [hunk 3]
```diff
@@ -208,6 +208,8 @@ class PlainTextHelper
         //  and not constrain text to a fixed-width column.
 
         'base_url' => '',
+
+        'preview_length' => 119, // Maximum length of the preview text
```
Comment (tomekkowalczyk) [inline app/bundles/EmailBundle/Helper/PlainTextHelper.php:212]: why do we have the value set to 119 characters?

**Review by tomekkowalczyk — CHANGES_REQUESTED** [review]
@patrykgruszka good job I left two comments

File: app/bundles/EmailBundle/Helper/PlainTextHelper.php (hunk 3 above)
Comment (patrykgruszka) [reply app/bundles/EmailBundle/Helper/PlainTextHelper.php:212]: Average character display for gmail webmail client. Based on https://www.emailonacid.com/blog/article/email-marketing/preview-vs-preheader-text-how-long-should-preheader-text-be/

**Comment by patrykgruszka** [comment]
@tomekkowalczyk thanks for CR, I've updated the code.

**Review by RCheesley — APPROVED** [review]
Thanks for the PR!

I tested this with the default and when changing to something bespoke (perhaps helpful to mention the preheader is displayed in the <mj-preview> in the code)

Before the PR:

screenshot-mautic ddev site-2024 05 29-09_59_39

With the PR, using the default which has {subject} in the mjml-title)

screenshot-mautic ddev site-2024 05 29-10_00_07

Changing the MJML-title field

screenshot-mautic ddev site-2024 05 29-10_06_39

Can you please make a PR to the docs to explain how to use the preheader and how it's shown? Thanks!

**Comment by escopecz** [comment]
We've actually built a new field to the email for the preheader. Let me try to push it.

**Comment by escopecz** [comment]
I started to push it to https://github.com/mautic/mautic/pull/13792 but it depends on some other PR so I'll try to push that as well

**Review by tomekkowalczyk — APPROVED** [review]

**Comment by patrykgruszka** [comment]
@escopecz that's a great feature! Ping me when it's ready to test. I think that this PR could be still merged to show the generated preview when the preheader from your PR isn't explicitly defined.

@RCheesley I'm not sure that there is much to describe in the docs for this preview. I've checked if there are any screenshots to update, but the email view isn't documented at all. It might be a good idea to consider adding some documentation alongside #13792, when it's ready. What do you think?

**Comment by RCheesley** [comment]
@patrykgruszka yes I think adding the docs alongside #13792 would be a great option.

**Pull request merged**